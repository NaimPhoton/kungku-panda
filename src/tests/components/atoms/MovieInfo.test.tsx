import React from "react";
import { render } from "@testing-library/react";
import MovieInfo, { MovieInfoProps } from "../../../components/atoms/MovieInfo";

describe("MovieInfo Component", () => {
  const defaultProps: MovieInfoProps = {
    label: "Test Label",
    value: "Test Value",
  };

  it("renders with the correct label and value", () => {
    const { getByText } = render(<MovieInfo {...defaultProps} />);

    // Assert that the component renders with the correct label and value
    expect(getByText(`${defaultProps.label}:`)).toBeInTheDocument();
    expect(getByText(defaultProps.value)).toBeInTheDocument();
  });

  it("matches snapshot", () => {
    const { asFragment } = render(<MovieInfo {...defaultProps} />);

    // Snapshot testing to ensure component structure remains consistent
    expect(asFragment()).toMatchSnapshot();
  });
});
