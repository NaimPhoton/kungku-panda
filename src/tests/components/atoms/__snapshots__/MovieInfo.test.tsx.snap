// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MovieInfo Component matches snapshot 1`] = `
<DocumentFragment>
  <div>
    <span>
      <strong>
        Test Label: 
      </strong>
    </span>
    <span>
      Test Value
    </span>
  </div>
</DocumentFragment>
`;
